spec/java/security/SecureRandom.main.lsl

```diff
@@ -0,0 +1,885 @@
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new LongStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.longs (@target self: SecureRandom, streamSize: long, randomNumberOrigin: long, randomNumberBound: long): LongStream
+    {
+        var size: int = streamSize as int;
+        if (size > MAX_RANDOM_STREAM_SIZE)
+            size = MAX_RANDOM_STREAM_SIZE;
+
+        val mass: array<long> = _generateRandomLongArrayWithBounds(size, randomNumberOrigin, randomNumberBound);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new LongStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = size,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.nextBoolean (@target self: SecureRandom): boolean
+    {
+        result = action SYMBOLIC("boolean");
+    }
+
+
+    fun *.nextBytes (@target self: SecureRandom, bytes: array<byte>): void
+    {
+        _nextBytes(bytes);
+    }
+
+
+    @Phantom proc nextBytes_loop (i: int, bytes: array<byte>): void
+    {
+        bytes[i] = action SYMBOLIC("byte");
+    }
+
+
+    // within java.util.Random
+    fun *.nextDouble (@target self: SecureRandom): double
+    {
+        result = action SYMBOLIC("double");
+
+        action ASSUME(0.0 <= result);
+        action ASSUME(result < 1.0);
+    }
+
+
+    // within java.util.Random
+    fun *.nextFloat (@target self: SecureRandom): float
+    {
+        result = action SYMBOLIC("float");
+
+        action ASSUME(0.0f <= result);
+        action ASSUME(result < 1.0f);
+    }
+
+
+    // within java.util.Random
+    @synchronized fun *.nextGaussian (@target self: SecureRandom): double
+    {
+        result = action SYMBOLIC("double");
+        val isNaN: boolean = action DEBUG_DO("Double.isNaN(result)");
+        action ASSUME(isNaN == false);
+    }
+
+
+    // within java.util.Random
+    fun *.nextInt (@target self: SecureRandom): int
+    {
+        result = action SYMBOLIC("int");
+    }
+
+
+    // within java.util.Random
+    fun *.nextInt (@target self: SecureRandom, bound: int): int
+    {
+        if (bound <= 0)
+            action THROW_NEW("java.lang.IllegalArgumentException", ["bound must be positive"]);
+
+        result = action SYMBOLIC("int");
+
+        action ASSUME(0 <= result);
+        action ASSUME(result < bound);
+    }
+
+
+    // within java.util.Random
+    fun *.nextLong (@target self: SecureRandom): long
+    {
+        result = action SYMBOLIC("long");
+    }
+
+
+    @synchronized fun *.setSeed (@target self: SecureRandom, seed: array<byte>): void
+    {
+        action DO_NOTHING();
```

Comment:
@dog-m As I understand - no;

```
    /**
     * Reseeds this random object. The given seed supplements, rather than
     * replaces, the existing seed. Thus, repeated calls are guaranteed
     * never to reduce randomness.
     *
     * @param seed the seed.
     *
     * @see #getSeed
     */
    synchronized public void setSeed(byte[] seed) {
        secureRandomSpi.engineSetSeed(seed);
    }
```

For example in secureRandomSpi can be such code:
```
if (seed == null)
    generateOwnSpecialSeed();
```

Reply:
As I can see it from existing implementations like:
- `sun.security.provider.SecureRandom#engineSetSeed` or
- `sun.security.pkcs11.P11SecureRandom#engineSetSeed`

it *will* cause NullPointerException to be thrown.